cmd/virt-launcher/qemu-kube

```diff
@@ -40,44 +41,50 @@ shift
 done
 
 if [ -z "$VM_NAME" ]; then
+    log "Libvirt is asking for qemu capabilities." >> $LOG
     exec $QEMU $ARGS
 fi
 
 if [[ "$VM_NAME" == *"="* ]]
 then
     VM_NAME=$(echo -n "$VM_NAME" | awk -F "=" -v RS="," "/guest/ { print \$2; }")
 fi
+log "${VM_NAME}: Trying to start VM ${VM_NAME}." >> $LOG
 
 function find_container() {
-    docker ps --no-trunc | awk "/.*compute.*virt-launcher-$VM_NAME-{5}.*/ { print \$1 }"
+    docker ps --no-trunc | awk "/.*compute.*virt-launcher-$VM_NAME-{5}.*/ { print \$1 }" 2>>$LOG
 }
 
 NEXT_WAIT=0
 until test -n "$(find_container)" || [ $NEXT_WAIT -eq 10 ]; do
+           log "${VM_NAME}: Waiting for container to show up." >> $LOG
            (( NEXT_WAIT++ ))
            sleep 6
 done
 
 CONTAINER_ID=$( find_container )
 
 if [ -z "$CONTAINER_ID" ]; then
+        log "Could not find the VM container" >> $LOG
         >&2 echo "Could not find the VM container"
         exit 1
 fi
 
+log "${VM_NAME}: Found container $CONTAINER_ID" >> $LOG
 CONTAINER_PID=$(docker inspect --format '{{.State.Pid}}' $CONTAINER_ID)
+
 CMD="$QEMU $ARGS"
 
-date >> $LOG
-echo "$CMD" >> $LOG
+log "${VM_NAME}: Container target pid is \"$CONTAINER_PID\"" >> $LOG
+
+log "${VM_NAME}: $CMD" >> $LOG
+
+set -e
 
 # Start qemu in the pid namespace of the container to receive signals on container kills
-# Don't close file descriptors bigger than 10000 to allow passing tap device fds
+# Don't close file descriptors smaller than 10000 to allow passing tap device fds
 # Start the qemu process in the cgroups of the docker container
-# to adher the resource limitations of the container.
+# to adhere to the resource limitations of the container.
 sudo -C 10000 unshare --mount bash -s << END
-  nsenter -m -t $CONTAINER_PID cat /etc/resolv.conf > /etc/resolv.conf.\$\$
```

Comment:
I am pretty sure we need this trick also in the container. AFAIK, if you are using the host network, you get the resolv.conf from the host (or at least inspired by the host network). I never saw the kubernetes dns entries in such daemon set. 

Did you already try it out with a disk attached?